metals/src/main/scala/scala/meta/internal/metals/UserConfiguration.scala

```diff
@@ -205,6 +215,28 @@ object UserConfiguration {
             Some(value)
         }
       }
+
+    def getStringListKey(key: String): Option[List[String]] =
+      getKey[List[String]](
+        key,
+        { elem =>
+          if (elem.isJsonArray()) {
+            val parsed = elem.getAsJsonArray().asScala.flatMap { value =>
+              Try(value.getAsJsonPrimitive().getAsString()) match {
+                case Failure(exception) =>
+                  errors += s"json error: values in '$key' should have value of type string but obtained $value"
+                  None
+                case Success(value) =>
+                  Some(value)
+              }
+            }
+            Option(parsed.toList)
+          } else {
+            None
```

Comment:
It doesn't matter much but I would check if it actually has any value or not and if it has but is not JsonArray I would do `errors += ...`

It might be helpful to catch potential error when other clients will start sending this value but will serialize it e.g. as String not array.

 But this is super edge case, you can completely ignore this.

Reply:
Added an additional error when we don't have a list.